arelle/SocketUtils.py

```diff
@@ -0,0 +1,17 @@
+import socket
+
+INTERNET_CONNECTIVITY = 'internetConnectivity'
+OFFLINE = 'offline'
+
+
+class WarnSocket(socket.socket):
+    """
+    This is a simple wrapper around the socket to print a warning if Arelle attempts to download somthing while running in offline mode.
+    """
+    def __init__(self, family: int = -1, type: int = -1, proto: int = -1, fileno: int | None = None):
```

Comment:
You'll need to add a `from __future__ import annotations` import at the top of this module for the `|` type union syntax to work with Python 3.8 and 3.9.